scripts/gcs/flask_gcs.py

```diff
@@ -8,221 +8,241 @@
-        return self.img_proc_status.value
+	def __init__(self):
+		self.manager = multiprocessing.Manager()
+		self.sda_status = self.manager.Value('s', "disconnected")
+		self.img_proc_status = self.manager.Value('s', "disconnected")
+		self.sda_start_time = self.manager.Value('i', int(datetime.utcnow().strftime("%s")))
+		self.img_proc_start_time = self.manager.Value('i', int(datetime.utcnow().strftime("%s")))
+		self.targets_to_submit = self.manager.list()
+
+		self.waypoints_reset = False
+
+		self.interop_client = self.manager.list()
+		self.interop_client.append(SUASSystem.InteropClientConverter())
+		self.interop_data = self.manager.list()
+		self.interop_data.append(self.get_interop_data())
+
+		self.gcs_process = multiprocessing.Process(target=SUASSystem.gcs_process, args=(
+			self.sda_status,
+			self.img_proc_status,
+			self.interop_client,
+			self.targets_to_submit
+		))
+		self.gcs_process.start()
+
+	def set_waypoint_reset(self):
+		if self.waypoints_reset is False:
+			self.waypoints_reset = True
+
+	def get_waypoint_reset(self):
+		return self.waypoints_reset
+
+	def get_interop_data(self):
+		try:
+			active_interop_mission = self.interop_client[0].get_active_mission()
+			obstacles = self.interop_client[0].get_obstacles()
+			active_interop_mission_json = SUASSystem.get_mission_json(active_interop_mission, obstacles)
+
+			data = {
+				"status": "connected",
+				"emergent_position": [
+					active_interop_mission_json["emergent_last_known_pos"]["latitude"],
+					active_interop_mission_json["emergent_last_known_pos"]["longitude"]
+				],
+				"airdrop_position": [
+					active_interop_mission_json["air_drop_pos"]["latitude"],
+					active_interop_mission_json["air_drop_pos"]["longitude"]
+				],
+				"off-axis_position": [
+					active_interop_mission_json["off_axis_target_pos"]["latitude"],
+					active_interop_mission_json["off_axis_target_pos"]["longitude"]
+				]
+			}
+		except:
+			data = {
+				"status": "disconnected",
+				"emergent_position": [0, 0],
+				"airdrop_position": [0, 0],
+				"off-axis_position": [0, 0]
+			}
+
+		return data
+
+	def set_sda_status(self, status):
+		if status == "connected":
+			self.sda_start_time.value = int(datetime.utcnow().strftime("%s"))
+
+		self.sda_status.value = status
+
+	def get_sda_status(self):
+		return self.sda_status.value
+
+	def set_img_proc_status(self, status):
+		if status == "connected":
+			self.img_proc_start_time.value = int(datetime.utcnow().strftime("%s"))
+
+		self.img_proc_status.value = status
+
+	def get_img_proc_status(self):
+		return self.img_proc_status.value
 
 app = Flask(__name__, static_url_path='')
 client = Client()
 
 @app.route('/')
 def index():
-    try:
-        return render_template('index.html', client=client)
-    except:
-        traceback.print_exc()
+	try:
+		return render_template('index.html', client=client)
+	except:
+		traceback.print_exc()
+
+@app.route('/get/waypoints/reset') # !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Comment:
This definition is not correct. You need to add a methods parameter here.

Reply:
No method parameters are defined because the API specification in https://github.com/FlintHill/SUAS-Competition/issues/106#issuecomment-362670527 does not specify request parameters since none are needed. I don't understand what you would pass.